### DISCUSSION TITLE
Timeout error

### DISCUSSION THREAD

**Discussion by Nickulous** [thoukydides/homebridge-homeconnect#133] [General]
I have an timeout error when I click on link to authorise the plugin. It calls me "Timeout error". Does anyone solve it?
<img width="742" alt="image" src="https://user-images.githubusercontent.com/48156613/224397176-b27d8217-926c-404c-b10b-dfb88f710263.png">

**Comment by thoukydides** [comment]
That message seems reasonably self explanatory, i.e. the authorisation link that you used was too old.

Each authorisation link expires 10 minutes after it is generated. Assuming that you are using a released version of this plugin (rather than the latest code in the github repo) a new authorisation link will be generated each time it expires. Make sure that you use the most recent link that appears in the log, rather than an earlier link. (Or if you are using homebridge-config-ui-x, and clicking on the link from the configuration editor, then close the configuration and re-open it to see the latest link.)

Note that the plugin must be running during the authorisation process.

**Comment by Nickulous** [comment]
I'm using last link... Еhe last time I waited until the link was updated and immediately followed it, but the error is always the same
<img width="281" alt="image" src="https://user-images.githubusercontent.com/48156613/224472435-8ac6cd8f-69da-46d9-ba48-68026885f11a.png">
<img width="702" alt="image" src="https://user-images.githubusercontent.com/48156613/224472490-1ca0b292-eb31-4714-b9cc-897c45a53949.png">
<img width="587" alt="image" src="https://user-images.githubusercontent.com/48156613/224472482-0f089455-4f2f-4c1d-8dcb-3707f66a55ed.png">

**Comment by thoukydides** [comment]
The applicaton looks correctly configured.

When is the error message displayed, e.g. immediately after clicking on the authorisation link, or after logging in with your SingleKey ID?

Have you tried using the log file instead of the configuration GUI? That may have logged other errors that give a hint as to what might be going wrong. It is also easier to see when a new authorisation link gets generated.

**Reply by Nickulous** [reply]
After typing my email address, I have no field for the passowrd.
Yes, I have tried it, but there Is nothing new... 
<img width="919" alt="image" src="https://user-images.githubusercontent.com/48156613/224473980-67885193-c3c3-4924-96e9-988ec503ab83.png">

**Reply by Nickulous** [reply]
and yes... I dont know how to link my SingleKey ID with HomeConnect account, maybe its a reson I have an error?

**Reply by Nickulous** [reply]
<img width="734" alt="image" src="https://user-images.githubusercontent.com/48156613/224474265-b6e5d079-7ec0-4a60-a19b-c19ee28df944.png">
This is the window after i click on auth link

**Comment by thoukydides** [comment]
It looks like the problem is that your email address isn't considered valid by the Home Connect authorisation servers. I suspect that you need to switch to using a SingleKey ID instead, and access the terms and conditions which were updated recently. You should be able to do both of those within the official Home Connect app on your iPhone.

**Reply by Nickulous** [reply]
> It looks like the problem is that your email address isn't considered valid by the Home Connect authorisation servers. I suspect that you need to switch to using a SingleKey ID instead, and access the terms and conditions which were updated recently. You should be able to do both of those within the official Home Connect app on your iPhone.

I cant find this thing in iPhone HomeConnect app...

**Comment by thoukydides** [comment]
From the information displayed within the Home Connect app…

> ## How to switch
> Enable SingleKey ID – it’s quick and easy.
> Step 1: Tap the profile icon in the app and go to [Personal Data](homeconnect://openPersonalData). A pop-up will direct you to the SingleKey ID registration website*.
> Step 2: Register your new SingleKey ID.
> Step 3: Connect your new ID to your Home Connect account.
…
> ## Do you already have a SingleKey ID?
> If you haven’t connected your existing Home Connect account to your SingleKey ID yet, simply choose the option “Log in with Home Connect”. Once you have linked your Home Connect account to your new SingleKey ID, you’re good to go. Please note that you will not be able to use your old Home Connect login anymore from that point.
> 
> *In case the pop-up doesn't come up in your profile, please check again in a few days.

**Comment by thoukydides** [comment]
You can find more help here:
https://www.home-connect.com/global/help-support/singlekey-id

**Comment by Nickulous** [comment]
Is there any option to make it working without SingleKey ID?

**Comment by thoukydides** [comment]
That's a question for the Home Connect team. They are the only ones who know how their authentication servers work...
https://www.home-connect.com/global/service/contact-customer-service

**Comment by Nickulous** [comment]
Ok, I figured out! Deleted my account and created a new one again, all works now!
I have another question, in your table the is a "Notify"options for Dishwasher, how can i create a notification for example if my salt is on low level?

**Comment by thoukydides** [comment]
Good to hear that you managed to sort out the authorization. The migration to SingleKey IDs does appear to have (unsurprisingly) caused quite a few problems.

Within the Home app you should that the dishwasher has "4 unconfigured buttons". These are:
1. Button 1: Program Finished
2. Button 2: Program Aborted
3. Button 3: Salt Low
4. Button 4: Rinse Aid Low

Next to each button it says "Add". Clicking on that will guide you through creating an HomeKit automation, e.g. to turn lights on. You cannot use it to create a notification (but the Home Connect app already provides those notifications, so there wouldn't be much point anyway).

If you use a third-party HomeKit app, such as Elagto's Eve, then it show the names for those buttons rather than just their numeric index.
